mds/apis/agency_api/v0_x/vehicles.py

```diff
@@ -331,8 +342,7 @@ def create(self, *args, **kwargs):
     @action(detail=True, methods=["post", "options"])
     def event(self, request, id):
         """Endpoint to receive an event from a provider."""
-        provider_id = request.user.provider_id
-        device = models.Device.objects.filter(provider_id=provider_id, id=id).last()
+        device = models.Device.objects.get(pk=id)
```

Comment:
This will raise an exception if the device is unknown, which leads to a 500 error. Check https://devdocs.io/django~2.1/topics/http/shortcuts#django.shortcuts.get_object_or_404 (maybe DRF has its own one).